index.js

```diff
@@ -112,6 +122,43 @@ class Conf {
 		}
 	}
 
+	_migrate(migrations = {}, versionToMigrate = '0.0.0') {
+		const MIGRATION_KEY = '__conf-migrated-version__';
+
+		const previousMigratedVersion = this.get(MIGRATION_KEY, '0.0.0');
+
+		const olderVersionsWithMigrations = Object.keys(migrations)
+			.filter(version => this._shouldVersionPerformMigration(version, previousMigratedVersion, versionToMigrate))
+			.sort(semver.compare);
+
+		const migrationsToRun = olderVersionsWithMigrations.map(version => migrations[version]);
+
+		migrationsToRun.forEach(migration => migration(this));
+
+		this.set(MIGRATION_KEY, versionToMigrate);
+	}
+
+	/**
+	 * Specifies if a version is suitable to run its migrations.
+	 * To be suitable, a version should be comprehended between the previous migrated version and the version desired to migrate.
+	 *
+	 * @param {string} version The version that's being tested to be suitable or not.
+	 * @param {string} previousMigratedVersion The version matching the last migration that ocurred.
+	 * @param {string} versionToMigrate The version that specifies what to migrate to (a.k.a the current version).
+	 * @returns {boolean} If the version is suitable or not.
+	 */
+	_shouldVersionPerformMigration(version, previousMigratedVersion, versionToMigrate) {
+		if (semver.lte(version, previousMigratedVersion)) {
+			return false;
+		}
+
+		if (semver.gt(version, versionToMigrate)) {
```

Comment:
Shouldn't this be `semver.gte`?

Reply:
Version = version currently being iterated
VersionToMigrate = version the client has

Let's say we have these migrations:

```js
const migrations = {
    '1.0.0': () => initialMigration
    '1.0.1': () => finalMigration
}
```

And the user version is `1.0.1`


Do we want to run :
1. `initialMigration` + `finalMigration` 
or
2.  Just `initialMigration`?

Your suggestions results in option #2, while the current code results in option #1.

Reply:
Please ignore my comment, the conditions seem to be correct. I guess it's late and I need to sleep 😴

Reply:
It's fine, review it when you have some coffee in your system 😄